File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int min = nums[0];
        int left = 0 , right = nums.size()-1;
        while (nums[left]!=nums[right]){
            if(nums[left]>nums[right]) left++;
            else if(nums[left]<nums[right]) right--;
        }
        return nums[left];
        
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l=0
        r=len(nums)-1
        mini = float('inf')

        while l<=r :
            m = (l+r) // 2

            mini = min(mini,nums[m])
            
            if nums[m]>nums[r]:
                l = m+1
            else:
                r = m-1

        return mini

